python/dgl/distributed/dist_graph.py

```diff
@@ -1091,25 +1089,41 @@ def _split_even(partition_book, rank, elements):
 
     # compute the offset of each split and ensure that the difference of each partition size
     # is 1.
-    part_size = len(eles) // num_clients
-    sizes = [part_size] * num_clients
-    remain = len(eles) - part_size * num_clients
-    if remain > 0:
-        for i in range(num_clients):
-            sizes[i] += 1
-            remain -= 1
-            if remain == 0:
-                break
-    offsets = np.cumsum(sizes)
+    offsets = _even_offset(len(eles), partition_book.num_partitions())
     assert offsets[-1] == len(eles)
 
-    if rank == 0:
-        return eles[0:offsets[0]]
-    else:
-        return eles[offsets[rank-1]:offsets[rank]]
+    # Get the elements that belong to the partition.
+    partid = partition_book.partid
+    part_eles = eles[offsets[partid] : offsets[partid + 1]]
 
+    return part_eles
 
-def node_split(nodes, partition_book=None, ntype='_N', rank=None, force_even=True):
+def _split_random_within_part(partition_book, rank, part_eles):
+    # If there are more than one client in a partition, we need to randomly select a subset of
+    # elements in the partition for a client. We have to make sure that the set of elements
+    # for different clients are disjoint.
+
+    num_clients = role.get_num_trainers()
+    num_client_per_part = num_clients // partition_book.num_partitions()
```

Comment:
If num_clients is 6, partitions is 4, then num_client_per_part = 1.
Will this cause any problem?

Reply:
this shouldn't happen. Each machine should have the same number of trainers. But we should add an assert to verify it.

Reply:
Did you add the assertion in somewhere?